### PR TITLE
HAL_F4Light: new exception handling

### PR DESCRIPTION
**Pull request by night-ghost**
kill task in case of exception in armed state (not in main task) instead of dead loop, or reboot a FC. Inspired by [this report](https://www.rcgroups.com/forums/showpost.php?p=39488195&postcount=2515)

### CODE REVIEW

File: libraries/AP_HAL_F4Light/0_Done.md [hunk 1]
```diff
@@ -156,6 +156,7 @@
 * renamed board *_MP32V1F4 to *_Revolution to simplify things
 * fixed bug in RC_Input that cause permanent Failsafe
 * added motor clipping reporting and baro compensation by GPS from https://github.com/DuraCopter/ardupilot
+* in case of any HardFault or Panic() in armed state kill a current task and resume (or reboot a FC) instead of hang
```
Comment (amilcarlucas) [inline libraries/AP_HAL_F4Light/0_Done.md]: add a comma between state and kill

File: libraries/AP_HAL_F4Light/boards/f4light_MatekF405_OSD/system_stm32f4xx.c [hunk 2]
```diff
@@ -333,7 +333,7 @@ void SystemCoreClockUpdate(void)
   SystemCoreClock >>= tmp;
 }
 
-extern void __error(uint32_t num, uint32_t pc, uint32_t lr);
+extern void __error(uint32_t num, uint32_t pc, uint32_t lr, uint32_t flag, uint32_t flag);
```
Comment (amilcarlucas) [inline libraries/AP_HAL_F4Light/boards/f4light_MatekF405_OSD/system_stm32f4xx.c]: copy, paste twice ?

Comment (night-ghost) [reply libraries/AP_HAL_F4Light/boards/f4light_MatekF405_OSD/system_stm32f4xx.c]: sure! The strange thing - there is no even GCC warning...

File: libraries/AP_HAL_F4Light/0_Done.md (hunk 1 above)
Comment (night-ghost) [reply libraries/AP_HAL_F4Light/0_Done.md]: sorry for my english :(

**Comment by tridge** [comment]
this needs a rebase as it conflicts with a previously merged PR

**Comment by night-ghost** [comment]
done

**Pull request merged**